### ISSUE TITLE
Fix 'resource != null && resource.exists()' pattern

### ISSUE THREAD

**Issue by briandealwis** [GoogleCloudPlatform/google-cloud-eclipse#3187]
@chanseokoh pointed out that the null check in code like the following should be unnecessary as the `getFile()` and `getFolder()` methods should never return null.
```java
IFile resource = project.getFile("name");
if(resource != null  && resource.exists()) {
```

What's worse is that we have some places that do cascading `getFolder()` + `getFile()` checks where they could be combined into a single line and single `exists()` check.

I believe we added these null tests because of mocks. We should fix the tests and mocks to adhere to the contracts.

**Comment by elharo** [comment]
Can you link to the documentation that promises these methods don't return null?

**Comment by briandealwis** [comment]
[`IContainer#getFile()`](https://help.eclipse.org/2019-03/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/core/resources/IContainer.html#getFile(org.eclipse.core.runtime.IPath)) and [`getFolder()`](https://help.eclipse.org/2019-03/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/core/resources/IContainer.html#getFolder(org.eclipse.core.runtime.IPath)) return a handle and do not check for existence — indeed, the way to create a file is to get an `IFile` resource and then call [`IFile.create()`](https://help.eclipse.org/2019-03/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/core/resources/IFile.html#create(java.io.InputStream,boolean,org.eclipse.core.runtime.IProgressMonitor)).  Other methods on `IContainer` like [`findMember()`](https://help.eclipse.org/2019-03/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/core/resources/IContainer.html#findMember(java.lang.String)) explicitly return `null` if no such resource exists.

**Comment by briandealwis** [comment]
I did some investigation and honestly I don't think this is worth the effort.  There are a couple of sources of difficulty:

  - The `IFile`/`IFolder` interfaces are quite large and many methods have multiple forms.  For example, iFolder offers `getFile(String)` and `getFile(IPath)`, `getFolder(String)` and `getFolder(IPath)`.  Each of those methods must return a handle object for whom `exists()` must return true or false.  Mocking these methods is annoying.  And there are dozens more of similar methods.
  - There are other libraries or helper classes that *do* return `null`. So we have odd patterns where some code does a `file != null && file.exists()` whereas other parts just do `file.exists()`.
    - The WTP `IVirtualComponent` equivalents *do* return `null` if not found
    - Many of own helper classes, like `WebProjectUtil` and `AppEngineConfigurationUtil`, have methods for finding various files that return `null` if not found.  

If we wanted to go down this route, it might be better to extend our `ResourcesUtil` class with a `Optional<IFile> getFile(IContainer source, String... paths)` that would return an optional that is present if the file exists.  And a similar method for `getFolder(IContainer source, String... paths)`.  But I'm not convinced this is that helpful either.

An alternative is to create a lo-fi mock `IWorkspace`.  I remember trying this at one point but I got bogged down and put it aside.
